ngraph/core/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -0,0 +1,633 @@
+                    case Transform_mode::asymmetric:
+                        return [](float x_resized, float x_scale, float, float) {
+                            return x_resized / x_scale;
+                        };
+                        break;
+                    case Transform_mode::tf_half_pixel_for_nn:
+                        return [](float x_resized, float x_scale, float, float) {
+                            return (x_resized + 0.5f) / x_scale;
+                        };
+                        break;
+                    case Transform_mode::align_corners:
+                        return [](
+                            float x_resized, float, float length_resized, float length_original) {
+                            return length_resized == 1
+                                       ? 0
+                                       : x_resized * (length_original - 1) / (length_resized - 1);
+                        };
+                        break;
+                    default:;
+                    }
+                    return [](float x_resized, float x_scale, float, float) {
+                        return ((x_resized + 0.5f) / x_scale) - 0.5f;
+                    };
+                }
+            };
+
+            class InterpolateEvalHelper final
+            {
+            public:
+                InterpolateEvalHelper() = default;
+
+                InterpolateEvalHelper(const op::v4::Interpolate::InterpolateAttrs& attrs,
+                                      const Shape& input_data_shape,
+                                      const std::vector<int64_t>& axes,
+                                      const Shape& out_shape,
+                                      const std::vector<float>& scales)
+                    : m_get_nearest_pixel{attrs.nearest_mode}
+                    , m_get_original_coord{attrs.coordinate_transformation_mode}
+                    , m_interp_mode{attrs.mode}
+                    , m_antialias{attrs.antialias}
+                    , m_cube_coeff{attrs.cube_coeff}
+                    , m_input_data_shape{input_data_shape}
+                    , m_axes{axes}
+                    , m_out_shape{out_shape}
+                    , m_scales{scales}
+                {
+                }
+
+                ~InterpolateEvalHelper() = default;
+
+                float triangle_coeff(float dz);
+                std::array<float, 4> get_cubic_coeff(float s, float a);
+            private:
+                GetNearestPixel m_get_nearest_pixel;
+                GetOriginalCoordinate m_get_original_coord;
+                InterpolateMode m_interp_mode;
+                double m_cube_coeff;
+                bool m_antialias;
+
+                Shape m_input_data_shape;
+                std::vector<int64_t> m_axes;
+                Shape m_out_shape;
+
+                std::vector<float> m_scales;
+            };
+
+            template <typename T>
+            class InterpolateEval final
```

Comment:
Why do we need to create this complex class only to call evaluate?

Can we replace this class on the method(s)?

Reply:
In this case we will have 4 large template methods with identical long initialization of  functions `get_nearest_pixel`, `get_original_coord`, and some attributes.

Reply:
Now I am moving all non-template parts into methods of helper class.